clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -2116,19 +2118,27 @@ CGCallee ItaniumCXXABI::getVirtualFunctionPointer(CodeGenFunction &CGF,
                                                   llvm::Type *Ty,
                                                   SourceLocation Loc) {
 #ifdef INTEL_SYCL_OPAQUEPOINTER_READY
-  llvm::Type *TyPtr = CGM.GlobalsInt8PtrTy;
+  llvm::Type *PtrTy = CGM.GlobalsInt8PtrTy;
 #else  // INTEL_SYCL_OPAQUEPOINTER_READY
   llvm::Type *TyPtr = Ty->getPointerTo();
 #endif // INTEL_SYCL_OPAQUEPOINTER_READY
   auto *MethodDecl = cast<CXXMethodDecl>(GD.getDecl());
+#ifdef INTEL_SYCL_OPAQUEPOINTER_READY
+  llvm::Value *VTable = CGF.GetVTablePtr(This, PtrTy, MethodDecl->getParent());
+#else
   llvm::Value *VTable = CGF.GetVTablePtr(
       This, TyPtr->getPointerTo(), MethodDecl->getParent());
+#endif
 
   uint64_t VTableIndex = CGM.getItaniumVTableContext().getMethodVTableIndex(GD);
   llvm::Value *VFunc;
   if (CGF.ShouldEmitVTableTypeCheckedLoad(MethodDecl->getParent())) {
     VFunc = CGF.EmitVTableTypeCheckedLoad(
+#ifdef INTEL_SYCL_OPAQUEPOINTER_READY
+        MethodDecl->getParent(), VTable, PtrTy,
+#else
```

Comment:
Looks like to me that we can avoid this #ifdef if we do the rename of TyPtr to PtrTy.
Other than this, we have to keep all the other #ifdef anyway, due to call to ->getPoitnerTo() , BitCast etc.